mcm/json_layer/request.py

```diff
@@ -2980,7 +2980,7 @@ def do_datatier_selection(possible_inputs, __prev_outputs):
             for possible in possible_inputs:
                  for taskName, tier in reversed(__prev_outputs):
                     for t in tier:
-                        if t == possible:
+                        if t == possible or (t[-1].isdigit() and t[:-1] == possible):
```

Comment:
I think this kind of names and the proposed validation are fragile. I suggest using a regex capture instead to fetch both the data tier and the version number. It could be something like:

```suggestion
                        datatier_version = re.match(r"([A-Z]+)(\d+)?")
                        if datatier_version:
                            datatier, _ = datatier_version.groups()
                            if datatier == possible:
                                return "%soutput" % (t), taskName
```

Reply:
I hate regexp ...  and honestly don't understand the need for such a completed set of code.
it's actually not `datatier`, but loosely (through cmsdriver) the `eventcontent`